src/dodal/devices/bimorph_mirrors/CAENels_bimorph_mirror_interface.py

```diff
@@ -0,0 +1,273 @@
+        self._status_channels: list = []
+
+        # because self.status is compiled in a signal at instantiation, these have to be define in __init__:
+        self.wait_till_idle = partial(
+            self.wait_for_signal_value,
+            self.status,
+            Status.IDLE,
+            signal_range=[Status.IDLE, Status.BUSY],
+        )
+        self.wait_till_busy = partial(
+            self.wait_for_signal_value,
+            self.status,
+            Status.BUSY,
+            signal_range=[Status.IDLE, Status.BUSY],
+        )
+
+    def wait_for_signal_value(
+        self,
+        signal: EpicsSignal,
+        value: Any,
+        timeout: float = 10.0,
+        sleep_time: float = 0.1,
+        signal_range: Union[list, None] = None,
+    ):
+        """Wait for a signal to display given value. Default to polling time of 0.1 seconds, and timeout after 10.0.
+        If signal_range is non-none, an exception will be raised if the signal value does not exist in signal_range list.
+
+        Args:
+            signal: The signal to be read from
+            value: Once this value is read from signal, the function will return
+            timeout: Time before an exception is raised in seconds (default 10.0)
+            sleep_time: Polling time in seconds (default 0.1)
+            signal_range: Range of allowed values. If value outside of this is read, an exception will be thrown
+        """
+        stamp = time.time()
+        res = signal.read()[signal.name]["value"]
+
+        while res != value:
+            if signal_range is not None:
+                if res not in signal_range:
+                    raise Exception(
+                        f"Out of range: {signal} showing {res} not in {signal_range}"
+                    )
+
+            if time.time() - stamp > timeout:
+                raise Exception(f"Timeout waiting for {signal} to show {value}")
+
+            time.sleep(sleep_time)
+
+            res = signal.read()[signal.name]["value"]
+
+    def protected_read(self, signal: EpicsSignal):
+        """Waits for bimorph to be idle, then reads from signal.
+
+        Args:
+            signal: The signal to be read from
+
+        Returns:
+            A dictionary contaning value(s) read and other relevant data
+        """
+        await_value(self.status, Status.IDLE).wait()
+        return signal.read()
+
+    def protected_set(self, signal: EpicsSignal, value) -> SubscriptionStatus:
+        """Waits for bimorph to be idle, writes to signal.
+
+        Args:
+            signal: Signal to be written to
+            value: Value to be written to signal
+
+        Returns:
+            A SubscriptionStatus to track that the bimorph has entered its
+                post-operation idle state.
+        """
+        await_value(self.status, Status.IDLE).wait()
+        status = signal.set(value)
+        await_value(self.status, Status.BUSY).wait()
+        return status & await_value(self.status, Status.IDLE)
```

Comment:
Did you mean to have something like
`return signal.set(value) &= await_value(self.status, Status.IDLE)`
instead?